datadog/dogstatsd/base.py

```diff
@@ -399,8 +399,11 @@ def _send_to_server(self, packet):
             # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
             pass
         except (socket.error, socket.herror, socket.gaierror) as se:
-            log.warning("Error submitting packet: {}, dropping the packet and closing the socket".format(se))
-            self.close_socket()
+            if se.errno == 11:
```

Comment:
This should be `errno.EAGAIN`.
The number is different depending on the platform.
I'd also prefer to see that under a specific `socket.error` `except` statement to be clearer. Catching everything like this is done is not a good thing, we should make clear things a bit.

Reply:
Good point on `errno.EAGAIN`.

What do you mean by a specific `socket.error` `except` statement? I can create a separate statement but if we want to keep the logic the same for non-EAGAIN errors, I'd have to add the `close_socket` logic there as well.

Reply:
I mean:

```python
except socket.error:
   if not EAGAIN:
     close()
except (socket.herror, socket.gaierror):
     close()
```

in summary :)